### PR TITLE
CASMTRIAGE-3232: Uploading BMC Recovery Firmware into TFTP Servers

### PR DESCRIPTION
**Pull request by mbuchmann-hpe**
Added prerequisites for uploading BMC recovery firmware into tftp servers.  Added a note in the install procedures to skip uploading to tftp if FAS not installed and ready, and to run the step after that has been completed.

### CODE REVIEW

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 `cray-upload-recovery-images` is a utility for uploading the BMC recovery files for ChassisBMCs, NodeBMCs, and RouterBMCs to be served by the *cray-tftp* service. The tool uses the cray cli (*fas*, *artifacts*) and *cray-tftp* to download the s3 recovery images (as remembered by FAS) then upload them into the PVC that is used by *cray-tftp*.
 `cray-upload-recovery-images` should be run on every system.
 
+**Prerequisite:**
```
Comment (nrockershousen) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:6]:
```suggestion
## Prerequisites
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+1. CSM software has been installed
```
Comment (nrockershousen) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:7]: Assuming the order doesn't matter here, these steps can be changed from an ordered list to an unordered list.

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 3]
```diff
@@ -3,6 +3,11 @@
...
+2. Cray CLI software has been installed
```
Comment (nrockershousen) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:8]:
```suggestion
2. The Cray command line interface \(CLI\) tool is initialized and configured on the system
```
This is the verbiage used in other procedures.

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 2 above)
Comment (nrockershousen) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:7]:
```suggestion
1. Cray System Management (CSM) software is installed
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 4]
```diff
@@ -3,6 +3,11 @@
...
+3. Firmware has been loaded into FAS as part of the HPC Firmware Pack (HFP) install
```
Comment (nrockershousen) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:9]:
```suggestion
3. Firmware is loaded into FAS as part of the HPC Firmware Pack (HFP) install; refer to the *HPE Cray EX System HPC Firmware Pack Installation Guide* on the HPE Customer Support Center for more information
```

File: install/configure_administrative_access.md [hunk 5]
```diff
@@ -83,11 +83,12 @@ BMC/controller passwords.
 
       The Olympus hardware (NodeBMCs, ChassisBMCs, RouterBMCs) needs to have recovery firmware loaded to the cray-tftp server in case the BMC loses its firmware. The BMCs are configured to load a recovery firmware from a TFTP server. This procedure does not modify any BMC firmware, but only stages the firmware on the TFTP server for download in the event it is needed.
 
+      This step requires CMS software install, Cray CLI software install, and HPC Firmware Pack (HFP) install.  If these are not currently installed, this step will need to be skipped and run later in the install process.
```
Comment (nrockershousen) [inline install/configure_administrative_access.md:86]:
```suggestion
      This step requires the CSM software, Cray CLI, and HPC Firmware Pack (HFP) to be installed. If these are not currently installed, this step will need to be skipped and run later in the install process.
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 1 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 2 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 3 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 4 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 2 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: install/configure_administrative_access.md (hunk 5 above)
Comment (mbuchmann-hpe) [reply install/configure_administrative_access.md]: Updated

**Review by nrockershousen — APPROVED** [review]

File: install/configure_administrative_access.md [hunk 6]
```diff
@@ -83,11 +83,12 @@ BMC/controller passwords.
 
       The Olympus hardware (NodeBMCs, ChassisBMCs, RouterBMCs) needs to have recovery firmware loaded to the cray-tftp server in case the BMC loses its firmware. The BMCs are configured to load a recovery firmware from a TFTP server. This procedure does not modify any BMC firmware, but only stages the firmware on the TFTP server for download in the event it is needed.
 
+      This step requires the CMS software, Cray CLI, and HPC Firmware Pack (HFP) to be installed.  If these are not currently installed, this step will need to be skipped and run later in the install process.
```
Comment (mharding-hpe) [inline install/configure_administrative_access.md:86]:
```suggestion
      This step requires the CSM software, Cray CLI, and HPC Firmware Pack (HFP) to be installed. If these are not currently installed, then this step will need to be skipped and run later in the install process.
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 7]
```diff
@@ -3,6 +3,11 @@
 `cray-upload-recovery-images` is a utility for uploading the BMC recovery files for ChassisBMCs, NodeBMCs, and RouterBMCs to be served by the *cray-tftp* service. The tool uses the cray cli (*fas*, *artifacts*) and *cray-tftp* to download the s3 recovery images (as remembered by FAS) then upload them into the PVC that is used by *cray-tftp*.
 `cray-upload-recovery-images` should be run on every system.
 
+## Prerequisites
++ Cray System Management (CSM) software is installed
```
Comment (mharding-hpe) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:7]:
```suggestion
* Cray System Management (CSM) software is installed.
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 8]
```diff
@@ -3,6 +3,11 @@
...
++ The Cray Command Line Interface (CLI) tool is initialized and configured on the system
```
Comment (mharding-hpe) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:8]:
```suggestion
* The Cray Command Line Interface (CLI) tool is initialized and configured on the system.
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 9]
```diff
@@ -3,6 +3,11 @@
...
++ Firmware is loaded into FAS as part of the HPC Firmware Pack (HFP) install; refer to the *HPE Cray EX System HPC Firmware Pack Install Guide* on the HPE Customer Support Center for more information.
```
Comment (mharding-hpe) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:9]:
```suggestion
* Firmware is loaded into FAS as part of the HPC Firmware Pack (HFP) install; refer to the *HPE Cray EX System HPC Firmware Pack Install Guide* on the HPE Customer Support Center for more information.
```

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md [hunk 10]
```diff
@@ -3,6 +3,11 @@
...
+
 ### Procedure
```
Comment (mharding-hpe) [inline operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md:11]:
```suggestion
## Procedure
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Also, does this need PRs for 1.2 and main as well?

File: install/configure_administrative_access.md (hunk 6 above)
Comment (mbuchmann-hpe) [reply install/configure_administrative_access.md:86]: Thanks - updated

Comment (mbuchmann-hpe) [reply install/configure_administrative_access.md]: 1.2 and master PRs will be coming after finalizing these updates and PR 1459 and 1460 have been merged.

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 7 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 9 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

File: operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md (hunk 10 above)
Comment (mbuchmann-hpe) [reply operations/firmware/Upload_Olympus_BMC_Recovery_Firmware_into_TFTP_Server.md]: Updated

**Review by mharding-hpe — APPROVED** [review]

**Review by gbaker-hpe — APPROVED** [review]

**Pull request merged**